repair/featurize/featurizer.py

```diff
@@ -5,9 +5,11 @@
 class Featurizer:
     __metaclass__ = ABCMeta
 
-    def __init__(self, name):
```

Comment:
I am curious why we removed the `name` from the constructor and moved it to `specific_setup`?

Reply:
What we had before was overriding the constructor and hence it was not possible to call "learnable=True" etc from the constructor. It had to be explicitly defined for each featurizer. With this change we do not need to define the same constructor over and over again.